Makefile refactor

joeflack4 opened the pull request:
- build-mondo-ingest: Added missing dependencies
- Removed some unneeded goals
- Moved around some goals

matentzn commented:
Problems @joeflack4, I will list them 1 by 1:

- [ ] `make[3]: *** No rule to make target 'reports/omim_exclusion_reasons.ttl'.  Stop.`

joeflack4 commented:
@matentzn Maybe you weren't in `src/ontology/`? I ran it and got this:
```
make reports/omim_exclusion_reasons.ttl
...
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://github.com/monarch-initiative/omim/releases/latest/download/omim.ttl \
	annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-omim.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2023-03-15/component-download-omim.owl --annotation owl:versionInfo 2023-03-15 -o tmp/component-download-omim.owl.owl; fi
```
I checked that box in your comment but if you notice a problem that remains, just uncheck it and let me know.

matentzn commented:
@joeflack4 I think i fixed the problem:

https://github.com/monarch-initiative/mondo-ingest/commit/63c9c1e24deb3aad6c791bf7a63c6730b3f84899

I can imagine it worked for you because you somehow had something in the mirror/%.owl from previous runs. Can you double check my fix is correct?

joeflack4 commented:
@matentzn Seems to be working fine, but I'm gonna let it keep running. FYI did you mean to commit that change to the `release07032023` branch? That's the one I checked out.

I didn't delete any files beforehand. Let me know if you want me to do that, though I don't think I need to do that as I'm running from your most recent changs.

matentzn commented:
Sorry I pushed it now to the correct PR. 

This fix I made - does it not make total sense? mirror/%.owl which is currently in the dependencies cant work, since `mirror/omim.owl` is not a goal that exists in the Makfile. It can only work if you are lucky enough to have that file available locally, which is probably why you were able to run the makefile in me not?

joeflack4 commented:
Yeah that makes perfect sense. I do see this in the ODK `Makefile`, though, which I had thought was being run in these cases:

```make
$(MIRRORDIR)/%.owl: mirror-% | $(MIRRORDIR)
	if [ $(IMP) = true ] && [ $(MIR) = true ] && [ -f $(TMPDIR)/mirror-$*.owl ]; then if cmp -s $(TMPDIR)/mirror-$*.owl $@ ; then echo "Mirror identical, ignoring."; else echo "Mirrors different, updating." &&\
		cp $(TMPDIR)/mirror-$*.owl $@; fi; fi
```

I actually was under the impression that `mondo-ingest.Makefile` imported `Makefile`, but now that I'm looking at it, I don't see an import line. Do these two makefiles interact with each other in some way (if so, how?) or are they kinda independent? Maybe something to discuss on a call.

joeflack4 commented:
I just made a target called `mirror/%.owl` in `Makefile`, and made a goal that used that as a prereq in `mondo-ingest.Makefile`, and it was able to use `mirror/%.owl` as a prereq, but when I deleted the `mirror/%.owl` goal, it did not use `$(MIRRORDIR)/%.owl` as a prereq instead, which is not what I expected. During this experiment, I didn't pass one of our ontology names as `%`, I just used random characters.

matentzn commented:
@hrshdhgd 

```
ValueError: ['rdfs', 'sssom', 'owl', 'oio', 'semapv', 'rdf'] are used in the SSSOM mapping set but it does not exist in the prefix map
make[1]: *** [mondo-ingest.Makefile:369: ../mappings/mondo-sources-all-lexical.sssom.tsv] Error 1
```

 I think you are working on this code as well, right? Can you get this to work? If not, let me know and I will try to hack it :)

- [ ] Fix message: ValueError: ['rdfs', 'sssom', 'owl', 'oio', 'semapv', 'rdf'] are used in the SSSOM mapping set but it does not exist in the prefix map

hrshdhgd commented:
You need to recreate `merged.db` from `merged.owl` using `semsql make tmp/merged.db -P config/prefix.csv` and update `prefix.csv` to contain all prefixes mentioned in the error. I have made those changes [in my PR ](https://github.com/monarch-initiative/mondo-ingest/pull/208)and it all works now.

matentzn commented on src/ontology/mondo-ingest.Makefile:
```diff
@@ -353,15 +357,27 @@ signature_reports: $(ALL_MIRROR_SIGNTAURE_REPORTS) $(ALL_COMPONENT_SIGNTAURE_REP
 
-../mappings/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py
-	python $^ run tmp/merged.db -c metadata/mondo.sssom.config.yml -r config/mondo-match-rules.yaml -o $@
+../mappings/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py tmp/merged.db
```
@hrshdhgd This here already ensures that `tmp/merged.db` is recreated!

hrshdhgd replied:
```
tmp/merged.db: tmp/merged.owl
	@rm -f .template.db
	@rm -f .template.db.tmp
	RUST_BACKTRACE=full semsql make $@
	@rm -f .template.db
	@rm -f .template.db.tmp
```
needs to be 
```
tmp/merged.db: tmp/merged.owl
	@rm -f .template.db
	@rm -f .template.db.tmp
	RUST_BACKTRACE=full semsql make $@ -P config/prefixes.csv
	@rm -f .template.db
	@rm -f .template.db.tmp
```

matentzn replied:
But these are all built-in prefixes - it should never be necessary to pass these in?

hrshdhgd replied:
You're right...

matentzn commented on src/ontology/mondo-ingest.Makefile:
```diff
@@ -300,24 +312,16 @@ deploy-mondo-ingest:
+		cd mondo/src/ontology &&\
+		make mondo.owl IMP=false MIR=false &&\
+		cd ../../../../ &&\
+		cp $(TMPDIR)/mondo/src/ontology/mondo.owl $@; fi
```
Feel free to add

joeflack4 replied:
Done!

matentzn commented:
```
python3 ../scripts/mirror_signature_via_oak.py --db-path components/ncit.db --onto-config-path metadata/ncit.yml --outpath reports/mirror_signature-ncit.tsv;\

Traceback (most recent call last):
  File "/work/src/ontology/../scripts/mirror_signature_via_oak.py", line 13, in <module>
    from oaklib import get_implementation_from_shorthand
  File "/usr/local/lib/python3.10/dist-packages/oaklib/__init__.py", line 9, in <module>
    from oaklib.selector import get_implementation_from_shorthand  # noqa:F401
  File "/usr/local/lib/python3.10/dist-packages/oaklib/selector.py", line 8, in <module>
    from oaklib.implementations import GildaImplementation
  File "/usr/local/lib/python3.10/dist-packages/oaklib/implementations/__init__.py", line 5, in <module>
    from oaklib.implementations.cx.cx_implementation import CXImplementation
  File "/usr/local/lib/python3.10/dist-packages/oaklib/implementations/cx/cx_implementation.py", line 6, in <module>
    import ndex2
ModuleNotFoundError: No module named 'ndex2'
make[3]: *** [mondo-ingest.Makefile:339: reports/mirror_signature-ncit.tsv] Error 1
make[3]: Leaving directory '/work/src/ontology'
make[2]: *** [mondo-ingest.Makefile:241: exclusions-ncit] Error 2
make[2]: Leaving directory '/work/src/ontology'
make[1]: *** [mondo-ingest.Makefile:246: exclusions-all] Error 2
make[1]: Leaving directory '/work/src/ontology'
make: *** [mondo-ingest.Makefile:300: build-mondo-ingest] Error 2
```

- [ ] Please include important dependencies in ODK - and always test that all pipelines run in ODK before merging (@joeflack4)
- [ ] Fix above error, thanks :)

hrshdhgd commented:
`ndex2` is a brand new `oaklib` dependency if that helps.

matentzn commented:
Thanks that helps. Joe ignore I have caused this by upgrading oak with --no-deps

joeflack4 commented:
That's what I was thinking when I glanced at it as well, that it was a new oak dependency introduced by us upgrading oak here. Thanks for taking a look at it Harshad!

matentzn commented:
@joeflack4 I assigned this to you know, getting

```
sh run.sh make dependencies build-mondo-ingest
```

To run is now the highest priority for us; I get too many roadblocks.

joeflack4 commented:
@matentzn Alright, will work on this. I thought the `dependencies` part is not something we typically want to include though? Is this just temporary until ODK is upgraded?

matentzn commented:
Exactly. I would be happy if it worked even without the dependency part!

matentzn commented:
- [ ] Next bug @joeflack4 (please always test the code with run.sh; this is the only way it will be run!)

```
make[2]: Leaving directory '/work/src/ontology'
python3 ../scripts/unmapped_tables.py \
--exclusions-path reports/omim_term_exclusions.txt \
--mondo-mappings-path tmp/mondo.sssom.tsv \
--onto-config-path metadata/omim.yml \
--db-path components/omim.db \
--outpath-simple reports/omim_unmapped_terms.tsv \
--outpath-full reports/omim_mapping_status.tsv
/usr/local/lib/python3.10/dist-packages/rdflib_jsonld/__init__.py:9: DeprecationWarning: The rdflib-jsonld package has been integrated into rdflib as of rdflib==6.0.0.  Please remove rdflib-jsonld from your project's dependencies.
  warnings.warn(
python3 ../scripts/deprecated_in_mondo.py \
--mondo-mappings-path tmp/mondo.sssom.tsv \
--mapping-status-path reports/omim_mapping_status.tsv \
--outpath reports/omim_mapped_deprecated_terms.robot.template.tsv
/usr/local/lib/python3.10/dist-packages/rdflib_jsonld/__init__.py:9: DeprecationWarning: The rdflib-jsonld package has been integrated into rdflib as of rdflib==6.0.0.  Please remove rdflib-jsonld from your project's dependencies.
  warnings.warn(
Traceback (most recent call last):
  File "/work/src/ontology/../scripts/deprecated_in_mondo.py", line 16, in <module>
    from src.scripts.utils import DOCS_DIR, REPORTS_DIR
ModuleNotFoundError: No module named 'src'
make[1]: *** [mondo-ingest.Makefile:411: reports/omim_mapped_deprecated_terms.robot.template.tsv] Error 1
rm components/omim.db
make[1]: Leaving directory '/work/src/ontology'
make: *** [mondo-ingest.Makefile:304: build-mondo-ingest] Error 2
Command exited with non-zero status 2
### DEBUG STATS ###
Elapsed time: 4:48:41
```

joeflack4 commented:
@matentzn My bad. Fixed the `ModuleNotFoundError: No module named 'src'` on `mapped-deprecated-terms-docs` in [this commit](https://github.com/monarch-initiative/mondo-ingest/commit/ef9f4c252a35fc4aeb52a63cefa478057555ee9d#diff-13bcf0b97f9e3a923033effa505263c2d1c07e6765f4a51b93bdb1a1182b80c5L16). Simple refactor error when I moved it from an `analysis` folder, I think. Surprised I didn't verify the refactor worked before.
